### PR TITLE
support SP2 in cloud7 as an option

### PR DESCRIPTION
**Pull request by bmwiedemann**
via want_sles12sp2=1

based on #999

### CODE REVIEW

**Comment by jsuchome** [comment]
+1 Looks correct to me

File: scripts/mkcloud [hunk 1]
```diff
@@ -638,6 +620,39 @@ function macfunc()
...
+    echo "$dist"
+}
+
+function get_admin_node_image_name()
+{
+    # get the name of the image to deploy the admin node
+    local dist=`get_admin_node_dist`
+    case $dist in
+        SLE12SP1) image=SLES12-SP1.qcow2 ;;
+        SLE12SP2) image=SLES12-SP2.qcow2 ;;
```
Comment (vuntz) [inline scripts/mkcloud]: Can we have a consistent order here? (SP2, SP1, GA, SLE11; or the opposite)

File: scripts/mkcloud [hunk 2]
```diff
@@ -638,6 +620,39 @@ function macfunc()
...
+{
+    # echo the name of the current dist for the admin node
+    local dist=
+    case $(getcloudver) in
+        7)  dist=SLE12SP1
+            [[ $want_sles12sp2 ]] && dist=SLE12SP2
+            ;;
+        6)  dist=SLE12SP1 ;;
+        5)  dist=SLE11
+            [[ $want_sles12 ]] && dist=SLE12
```
Comment (vuntz) [inline scripts/mkcloud]: Is this actually correct? Cloud 5 only supports SLE11 for the admin server.

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1135,9 +1210,12 @@ function onadmin_prepare_cloud_repos()
                 addsp3testupdates
                 addsles12testupdates
                 ;;
-            develcloud6|develcloud7|mitakacloud7|susecloud7|M?|Beta*|RC*|GMC*)
+            develcloud6)
                 addsles12sp1testupdates
                 ;;
+            develcloud7|mitakacloud7|susecloud7|M?|Beta*|RC*|GMC*)
+                addsles12sp2testupdates
+                ;;
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: Don't we need some `$want_sles12sp2` love here?

File: scripts/mkcloud (hunk 2 above)
Comment (bmwiedemann) [inline scripts/mkcloud]: I guess that was meant for the lonelynode codepath which now goes through onhost_deploy_image -> get_admin_node_image_name -> get_admin_node_dist.
but it seem to be wrong as is.

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: we dont have testupdates atm, so this is mostly a NOP. except if you wanted to test experimental SP1 updates, but we already cover those with cloud6

File: scripts/mkcloud (hunk 2 above)
Comment (bmwiedemann) [inline scripts/mkcloud]: pushed fixups to distinguish between lonely and admin case

**Comment by vuntz** [comment]
+1

**Pull request merged**